### PR TITLE
refactor: Step Info contract updates

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Update backend to align with contract changes RE `StepInfo`

JIRA: [AU-1480](https://2u-internal.atlassian.net/browse/AU-1480)

**What changed?**

- Change `activeStepInfo` to `stepInfo`
- Update sub-steps to include `closed` and `closedReason`
- Add some missing `PageContextSerializer` tests
- Add some utils for mapping and comparing workflow steps and progress

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled

**Testing Instructions**

- [ ] Unit tests.
- Exercise the following endpoints:
    - [ ] `get_block_learner_assessment_data`
    - [ ] `get_block_learner_submission_data` 

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Review by muselesscreator — COMMENTED** [review]
the back/forth between snake/camel-case is still a bit offputting for me (especially `get_closedReason`), but beyond that, lgtm.

**Comment by nsprenkle** [comment]
> the back/forth between snake/camel-case is still a bit offputting for me (especially `get_closedReason`), but beyond that, lgtm.

Oop, I didn't realize we *didn't* need to `camelCase` the keys (values we do need to `camelCase`). When we are close to getting a v1 out, I would vote for a renaming back to `snake_case` for Python prettiness

**Review by leangseu-edx — APPROVED** [review]
